src/string.rs

```diff
@@ -357,8 +357,8 @@ impl From<GString> for String {
         // since a lower value is always fine.
         // We also add one for the nul teminator.
         let capacity = len + 1;
-
-        unsafe { Self::from_raw_parts(s.into_raw().cast(), len, capacity) }
+        let ptr = s.as_ptr() as *mut u8;
```

Comment:
Won’t this result in use-after-frees, because at the end of this function the `GString` is dropped and thus the memory behind it is freed?

Reply:
I think we need [`.into_glib_ptr()`](https://docs.rs/glib/latest/glib/struct.GString.html#impl-IntoGlibPtr%3C*mut+i8%3E-for-GString) instead